protocol/x/clob/ante/clob.go

```diff
@@ -250,9 +256,93 @@ func IsShortTermClobMsgTx(ctx sdk.Context, tx sdk.Tx) (bool, error) {
 
 	numMsgs := len(msgs)
 	if numMsgs > 1 {
+		return true, errorsmod.Wrap(
+			sdkerrors.ErrInvalidRequest,
+			"a transaction containing short term MsgCancelOrder or MsgPlaceOrder may not contain more than one message",
+		)
+	}
+
+	return true, nil
+}
+
+// hasClobMsg returns `true` if the transaction has a single clob msg
+func HasClobMsg(tx sdk.Tx) bool {
+	msgs := tx.GetMsgs()
+
+	for _, msg := range msgs {
+		switch msg.(type) {
+		case *types.MsgCancelOrder:
+			return true
+		case *types.MsgPlaceOrder:
+			return true
+		case *types.MsgBatchCancel:
+
+			return true
+		}
+	}
+	return false
+}
+
+// IsValidClobMsgTx returns `true` if the transaction contains a valid set of clob msgs
+// A transaction with a clob msg must adhere to the below conditions
+//   - If the tx contains a short term order msg, the tx can only have one msg
+//   - If the tx contains a stateful order msg, it can only contain other stateful order msgs
+//     or a single transfer msg
+func IsValidClobMsgTx(tx sdk.Tx) (bool, error) {
```

Comment:
Current naming is confusing when it returns `false`: does it mean it's invalid or just doesn't contain clob message? Suggestion: rename to `IsClobTx`